shared/src/main/scala/amf/core/internal/plugins/syntax/StringDocBuilder.scala

```diff
@@ -0,0 +1,144 @@
+package amf.core.internal.plugins.syntax
+
+import amf.core.client.common.position.Position
+import amf.core.client.common.position.Position._
+import amf.core.client.scala.parse.document.{ParsedDocument, StringParsedDocument}
+import amf.core.internal.plugins.syntax.StringDocBuilder.INDENTATION_WIDTH
+import org.yaml.builder.DocBuilder
+
+import scala.collection.mutable
+
+class SourceCodeBlock(val indentation: Integer, val lines: mutable.Buffer[(String, Position)] = mutable.Buffer()) {
```

Comment:
Why does it receive a mutable.Buffer? Can't this be immutable?

Reply:
Check lines 61 and 66, is mutable